ui/client/components/RouteLeavingGuard.tsx

```diff
@@ -0,0 +1,77 @@
+import * as H from "history"
+import React, {useCallback, useEffect, useRef} from "react"
+import {useSelector} from "react-redux"
+import {Prompt} from "react-router"
+import * as DialogMessages from "../common/DialogMessages"
+import {isPristine} from "../reducers/selectors/graph"
+import {useWindows} from "../windowManager"
+
+interface RouteLeavingGuardProps {
+  when?: boolean,
```

Comment:
why `when` is nullable ? according to https://reactrouter.com/core/api/Prompt it can be either false or true